### PR TITLE
Create a check-example task for smarter example checking with PhantomJS

### PR DESCRIPTION
**Pull request by ahocevar**
This is a follow-up pull request to #5713, which avoids the problems mentioned there.

### CODE REVIEW

File: tasks/check-example.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/*eslint-env es6*/
+
+const http       = require('http');
+const path       = require('path');
+const serveFiles = require('serve-files');
+const spawn      = require('child_process').spawn;
```
Comment (tschaub) [inline tasks/check-example.js]: Watch out, we might add a linter rule that prohibits this sort of ASCII art in the future :).

File: tasks/check-example.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+/*eslint-env es6*/
```
Comment (tschaub) [inline tasks/check-example.js]: We could add `tasks/.eslintrc` with `{"env": {"es6": true}}` to get the same for all tasks.

File: tasks/check-example.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+['SIGHUP', 'SIGINT', 'SIGQUIT', 'SIGILL', 'SIGTRAP', 'SIGABRT',
+ 'SIGBUS', 'SIGFPE', 'SIGUSR1', 'SIGSEGV', 'SIGUSR2', 'SIGTERM'].forEach(signal => {
+   process.once(signal, () => {
+     process.stdout.write(`Got ${signal}, stopping...\n`),
+     server.close(() => {
+       process.stdout.write('Stopped.\n');
+       process.exit(0);
+     });
+
+     Object.keys(connections).forEach(cid => connections[cid].destroy());
```
Comment (tschaub) [inline tasks/check-example.js]: Surprised to see that this needs to be done.  What is the result if you don't destroy these?

**Comment by tschaub** [comment]
Didn't test it out locally, but this looks good to me.  Nice work tightening this up.

**Comment by tschaub** [comment]
@ahocevar I'm inclined to merge this since this is the only branch with passing tests.  I'd like to do some rebasing after this is in.  Hope that is ok by you.

**Comment by ahocevar** [comment]
Thanks @tschaub, that's perfectly fine. I'll create a follow-up that addresses your comments.

**Pull request merged**